### PR TITLE
Switch to Microsoft.Extensions.HostFactoryResolver.Sources

### PR DESCRIPTION
**Pull request by bricelam**

### CODE REVIEW

**Comment by bricelam** [comment]
Logging messages are a bit over-simplified, but they should still lead you down the right path when something goes wrong.

File: src/EFCore.Design/Properties/DesignStrings.resx [hunk 1]
```diff
@@ -241,23 +241,17 @@ Change your target project to the migrations project by using the Package Manage
...
   </data>
   <data name="FindingServiceProvider" xml:space="preserve">
     <value>Finding application service provider...</value>
   </data>
   <data name="NoBuildWebHost" xml:space="preserve">
-    <value>No CreateWebHostBuilder(string[]) method was found on type '{programClass}'.</value>
-  </data>
-  <data name="NoEntryPoint" xml:space="preserve">
-    <value>No entry point was found for assembly '{startupAssembly}'.</value>
+    <value>No CreateHostBuilder(string[]) method with return type 'IHostBuilder' was found on type 'Program'.</value>
```
Comment (Tratcher) [inline src/EFCore.Design/Properties/DesignStrings.resx]: This is a bit misleading, it assumes people have/will move to generic host.

Comment (bricelam) [reply src/EFCore.Design/Properties/DesignStrings.resx]: I'll do another pass on them. Will the new web templates look like this?
``` csharp
class Program
{
    public IWebHostBuilder CreateHostBuilder(string[] args);
}
```

Comment (Tratcher) [reply src/EFCore.Design/Properties/DesignStrings.resx]: https://github.com/aspnet/AspNetCore/pull/5706
`public static IHostBuilder CreateHostBuilder(string[] args);`

**Comment by Tratcher** [comment]
Better

**Pull request merged**